volunteer.php

```diff
@@ -650,4 +664,62 @@ function volunteer_civicrm_fieldOptions($entity, $field, &$options, $params) {
   if ($entity == "UFJoin" && $field == "entity_table" && $params['context'] == "validate") {
     $options[CRM_Volunteer_DAO_Project::getTableName()] = CRM_Volunteer_DAO_Project::getTableName();
   }
+}
+
+/**
+ * This function fetches the defaults from civicrm settings
+ * And puts them into the appropriate data format to return
+ * to the angular front-end
+ *
+ * @return array
+ * @throws CiviCRM_API3_Exception
+ */
+function volunteer_composeDefaultArray() {
+  $defaults = array();
+  $settings = CRM_Core_BAO_Setting::getItem("volunteer_defaults");
+
+  $defaults['is_active'] = (array_key_exists("volunteer_default_is_active", $settings)) ? $settings['volunteer_default_is_active'] : 1;
+  $defaults['campaign_id'] = (array_key_exists("volunteer_default_campaign", $settings)) ? $settings['volunteer_default_campaign'] : '';
+  $defaults['loc_block_id'] = (array_key_exists("volunteer_default_locblock", $settings)) ? $settings['volunteer_default_locblock'] : '';
+
+  $coreDefaultProfile = array(
```

Comment:
It feels like too much is going on with the profiles in this function. I'd argue that the defaults you're setting in `$coreDefaultProfile` belong in volunteer.setting.php.

Also, the foreach loop below overrides the database. Why not get the weight, for instance, from the `civicrm_uf_join` table instead of forcing one? Likewise, if in the future we load more into `module_data`, this code will have to be rewritten.

Finally, I haven't reviewed the PR in its entirety yet, but I haven't seen the reason for having three separate settings for the different uses of the profile (individual, group, both). If we add a new use, will we then need a new setting? Why not one ring to rule them all? Pseudocode:

```
var volunteerProjectDefaultProfile = [
  {
    use: 'primary',
    id: 25
  },
  {
    use: 'primary',
    id: 19
  },
  {
    use: 'both',
    id: 77
  }
];
```

... or ...

```
var volunteerProjectDefaultProfile = {
  primary: [1, 40, 12],
  additional: []
  both: [3]
};
```